File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/IotDeviceFault.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 设备故障实体，对应 iot_device_fault 表。
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class IotDeviceFault {

    private Long id;
    private Long deviceId;
    /** 查询时关联得到的设备所有者 ID，用于移动端权限判断。 */
    private Long deviceOwnerId;
    private String deviceCode;
    private String deviceName;
    private String plotName;
    private String faultCode;
    private String faultName;
    private Integer faultType;
    private Integer severity;
    private String faultDesc;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private Integer duration;
    private Integer status;
    private Integer assignStatus;
    private Long handleUserId;
    private String handleUsername;
    private String handleNickname;
    private LocalDateTime handleTime;
    private String handleResult;
    /** farm 端完成处理时上传的现场图片相对地址。 */
    private String completionImageUrl;
    /** 逻辑删除标记：0正常，1已删除。 */
    private Integer isDeleted;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: smart_farm/src/api/iotDeviceFault.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface IotDeviceFault {
  id?: number
  deviceId: number
  deviceCode?: string
  deviceName?: string
  plotName?: string
  faultCode: string
  faultName: string
  faultType: number
  severity?: number
  faultDesc?: string
  startTime?: string
  endTime?: string
  duration?: number
  status?: number
  assignStatus?: number
  handleUserId?: number
  handleUsername?: string
  handleNickname?: string
  handleTime?: string
  handleResult?: string
  createTime?: string
  updateTime?: string
}

export interface IotDeviceFaultPayload {
  deviceId: number
  faultCode: string
  faultName: string
  faultType: number
  severity?: number
  faultDesc?: string
  startTime?: string
  endTime?: string
  duration?: number
  status?: number
  assignStatus?: number
  handleUserId?: number
  handleTime?: string
  handleResult?: string
}

export interface UpdateIotDeviceFaultPayload extends Partial<IotDeviceFaultPayload> {
  id: number
}

export interface IotDeviceFaultListParams {
  deviceId?: number
  deviceName?: string
  faultCode?: string
  faultName?: string
  faultType?: number
  severity?: number
  status?: number
  handleUserId?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listIotDeviceFaults = (params: IotDeviceFaultListParams) => {
  return get<ApiResult<PageInfo<IotDeviceFault>>>(
    '/iot-device-fault/list',
    params as unknown as Record<string, unknown>,
  )
}

export const getIotDeviceFaultById = (id: number) => {
  return get<ApiResult<IotDeviceFault>>(`/iot-device-fault/${id}`)
}

export const addIotDeviceFault = (data: IotDeviceFaultPayload) => {
  return post<ApiResult<IotDeviceFault>>(
    '/iot-device-fault/add',
    data as unknown as Record<string, unknown>,
  )
}

export const updateIotDeviceFault = (data: UpdateIotDeviceFaultPayload) => {
  return http<ApiResult<IotDeviceFault>>({ url: '/iot-device-fault', method: 'put', data })
}

export const deleteIotDeviceFault = (id: number) => {
  return http<ApiResult<void>>({ url: `/iot-device-fault/${id}`, method: 'delete' })
}

export const batchDeleteIotDeviceFaults = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/iot-device-fault/batch', method: 'delete', data: ids })
}

export const updateIotDeviceFaultStatus = (id: number, status: number, handleResult?: string) => {
  return http<ApiResult<void>>({
    url: `/iot-device-fault/${id}/status`,
    method: 'put',
    params: handleResult ? { status, handleResult } : { status },
  })
}

export const assignIotDeviceFault = (id: number, handleUserId: number) => {
  return http<ApiResult<void>>({
    url: `/iot-device-fault/${id}/assign`,
    method: 'put',
    params: { handleUserId },
  })
}

export const acceptIotDeviceFaultAssignment = (id: number) => {
  return http<ApiResult<void>>({
    url: `/iot-device-fault/${id}/assignment/accept`,
    method: 'put',
  })
}

export const rejectIotDeviceFaultAssignment = (id: number, rejectReason?: string) => {
  return http<ApiResult<void>>({
    url: `/iot-device-fault/${id}/assignment/reject`,
    method: 'put',
    params: rejectReason ? { rejectReason } : undefined,
  })
}
